src/utils/BlockUtils.ts

```diff
@@ -40,13 +49,25 @@ export async function averageBlockTime(
     return { average: cache.average, blockRange: cache.blockRange };
   }
 
-  latestBlockNumber ??= await provider.getBlockNumber();
+  if (!isDefined(latestBlockNumber)) {
+    latestBlockNumber = await provider.getBlockNumber();
```

Comment:
Can you add a comment here that you are fetching the latest block from the primary provider, and then trying to reach quorum on it later on? This is why we add an offset to be safe in case the primary is lagging

Reply:
Done :+1: